File: my-folder/0234-palindrome-linked-list/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    bool isPalindrome(ListNode* head) {
        string s="";
        if(head==nullptr) return true;
        ListNode* newe = head;
        while(newe && newe->next!=head){
            s+=(char)newe->val;
            newe=newe->next;
        }
        string s2 = s;
        reverse(s2.begin(),s2.end());
        return s==s2;
        
    }
};

File: my-folder/0234-palindrome-linked-list/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    ListNode temp;
    boolean solve(ListNode p){
        if(p==null) return true;
        boolean ans = solve(p.next) && temp.val==p.val;
        temp = temp.next;
        return ans;
    }
    public boolean isPalindrome(ListNode head) {
        temp = head;
        return solve(head);
        
    }
}
